### PR TITLE
Compose env vars

### PR DESCRIPTION
**Pull request by matt-hensley**

### CODE REVIEW

**Review by MattFrick — APPROVED** [review]
Thanks, this adds so much functionality!
Eventually we should be able to remove the template parsing.
I also wonder if creating a single generated config that would "include" the template and regular docker-compose file(s) might be simpler.

File: yaml/model.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ type JavaGeneratorParams struct {
 type DockerCompose struct {
 	Generator           string              `yaml:"generator"`
 	Files               []string            `yaml:"files"`
+	Environment         []string            `yaml:"environment"`
```
Comment (zeitlinger) [inline yaml/model.go:59]:
```suggestion
	Environment         []string            `yaml:"env"`
```

Comment (zeitlinger) [reply yaml/model.go:59]: `env` matches k8s pods

File: yaml/generator.go [hunk 2]
```diff
@@ -78,6 +78,16 @@ func (c *TestCase) generateDockerComposeFile() []byte {
 	vars["LokiHTTPPort"] = c.PortConfig.LokiHTTPPort
 	vars["TempoHTTPPort"] = c.PortConfig.TempoHTTPPort
 
+	env := []string{}
+
+	for k, v := range vars {
+		env = append(env, "OATS_"+strings.ToUpper(k)+"="+v.(string))
```
Comment (zeitlinger) [inline yaml/generator.go:84]:
```suggestion
		env = append(env, k+"="+v.(string))
```

Comment (zeitlinger) [reply yaml/generator.go:84]: prepend OATS_ and make upper case in all usages - that makes it easier to follow

File: yaml/generator.go [hunk 3]
```diff
@@ -78,6 +78,16 @@ func (c *TestCase) generateDockerComposeFile() []byte {
...
+	}
+
+	for _, v := range c.Definition.DockerCompose.Environment {
+		env = append(env, "OATS_"+v)
```
Comment (zeitlinger) [inline yaml/generator.go:88]:
```suggestion
		env = append(env, k+"="+v)
```

Comment (zeitlinger) [reply yaml/generator.go:88]: we also may want to pass in environment variables to the application directly, so prepending OATS_ won't work.

**Review by zeitlinger — CHANGES_REQUESTED** [review]

**Comment by MattFrick** [comment]
> we also may want to pass in environment variables to the application directly, so prepending OATS_ won't work.
Is that case something like when a (Java) generator wants to set an env var that is also used outside of oats, so it can't have its own unique, oats-prefixed name?
If we don't generally prepend OATS_, could we maybe change the name of ConfigDir to OatsConfigDir?  That one in particular strikes me as really vague and other things might use the env var CONFIGDIR.

**Review by grcevski — APPROVED** [review]
LGTM!

**Review by zeitlinger — APPROVED** [review]

**Pull request merged**